select/index.js

```diff
@@ -176,27 +174,60 @@ class IrnmnSelect extends HTMLElement {
         const list = this.querySelector(`.${CLASS_NAMES.list}`);
         const header = this.querySelector(`.${CLASS_NAMES.header}`);
 
-        if (this.isOpen) {
-            this.determineDropdownPosition();
-            this.focusItem(
-                this.selectedOption !== null ? this.selectedOption : 0,
-            );
-        } else {
-            list.classList.remove(CLASS_NAMES.openUpwards);
-        }
-
         list.classList.toggle(CLASS_NAMES.listOpen, this.isOpen);
         header.setAttribute('aria-expanded', this.isOpen);
 
         if (this.isOpen) {
+            this.determineDropdownPosition();
+
+            const firstVisibleOptionIndex = this.findFirstVisibleOptionIndex();
+
             this.focusItem(
-                this.selectedOption !== null ? this.selectedOption : 0,
+                this.selectedOption !== null ? this.selectedOption : firstVisibleOptionIndex,
             );
         } else {
+            list.classList.remove(CLASS_NAMES.openUpwards);
             header.focus();
         }
     }
 
+    findVisibleOptionIndex(startIndex, direction) {
+        const options = this.querySelectorAll(`.${CLASS_NAMES.itemSelectable}`);
+        const totalOptions = options.length;
+    
+        // Loop through options, wrapping around using modular arithmetic
+        for (let i = 0; i < totalOptions; i++) {
```

Comment:
I suggest findIndex instead of for loop: [ options.findIndex](https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex)

Reply:
@samuelsilvapt this was a good idea but when i tried findIndex it didn't work very well with 'Circular Traversal' and made the code complicated.

From ChatGPT... _A for loop allows you to traverse the array in both directions with minimal adjustments_